_posts/2020-11-19-testy-e2e-cypress.md

```diff
@@ -0,0 +1,131 @@
+
+Wykonywane scenariusze testowe mogą być przez nas podglądane na żywo - są one uruchamiane na wybranej przez nas przeglądarce, może to być Edge, Chrome, Firefox lub wbudowany Electron. Podczas ich wykonywania każdy kolejny krok zapisywany jest pod postacią snapshotów - migawek, do których możemy zajrzeć w każdym momencie i zweryfikować stan aplikacji.
+   
+W Cypressie developer nie musi pamiętać aby pisać jawne oczekiwanie na zakończenie poleceń (_wait_ znane z Selenium). Jest to zrobione za nas przez twórców frameworka - wszystko dzieje się automatycznie i kolejne polecenia oraz asercje wykonują się w odpowiednim momencie.   
+
+Kolejną z rzeczy wartych wspomnienia jest łatwość użycia i konfiguracji - jedno polecenie instaluje framework, a kolejne uruchamia dashboard, którym posługiwanie się jest intuicyjne.
+
+<div class="img-with-legend">
+<img alt="Dashboard Cypress" src="/assets/img/posts/2020-11-19-testy-e2e-cypress/cypress-dashboard.png" />
+<span class="img-legend">Rys. 1. Dashboard Cypress</span>
+</div>
+
+# Rozpoczynanie pracy
+W kwestii wymagań, Cypress nie potrzebuje wiele: wystarczy Node.js oraz ulubione IDE.
+Aby umożliwić rozpoczęcie pracy, wystarczy wykonać następujące polecenia w katalogu projektu:
+1. `npm init`, aby stworzyć projekt node'owy
+2. `npm install cypress --save-dev` dla instalacji Cypressa
+
+Powstanie następująca domyślna struktura:
+<div class="img-with-legend">
+<img alt="Struktura Cypress" src="/assets/img/posts/2020-11-19-testy-e2e-cypress/struktura-cypress.png" />
+<span class="img-legend">Rys. 2. Struktura projektu Cypress</span>
+</div>
+Objaśnienia:
+- `fixtures` - miejsce gdzie możemy przechowywać gotowe zestaw danych, np. służące do zamockowania usług,
+- `integration` - lokalizacja testów,
+- `plugins` - miejsce do załączania zewnętrznych rozszerzeń dla Cypressa,
+- `support` - tutaj znajdą się np. stałe powtarzające się w wielu scenariuszach czy też nowe, customowe polecenia do globalnego reużycia.
+
+Następnie przy pomocy komendy `npx cypress open` możemy uruchomić dashboard i jeden z przykładowych testów.
+
+# Scenariusze testowe
+W ramach tego wpisu, na warsztat weźmiemy stronę główną bloga Consdata Tech -  [https://blog.consdata.tech/](blog.consdata.tech) i stworzymy dla niej dwa przypadki testowe.
+
+Pierwszy scenariusz będzie polegał na wejście na stronę, poczekaniu aż się załaduje i zweryfikowaniu kilku elementów, która potwierdzą nam, że portal jest w pełni działający.
+
+Drugi scenariusz, nieco bardziej rozbudowany, dokona kilku interakcji z blogiem.
+
+# Pierwszy test
+
+W katalogu `integration` stwórzmy nowy plik o nazwie `blog.spec.js`, przejdźmy do jego edycji i uzupełnijmy go o następującą treść:
```

Comment:
Zachęcam do stosowania jednej formy osobowej w instrukcjach. Odnosi się to do zwrotu "stwórzmy", zamieniłbym ją na "tworzymy". Następnie "przechodzimy" zamiast przejdźmy i "uzupełniamy" zamiast "uzupełnijmy".

Chyba, że stosowane w tym zdaniu formy są Ci bliższe, wówczas z kolei wróciłbym do wcześniejszych sformułowań i tamtych zastosował te zachęcająco brzmiące formy.